apps/api/src/services/data-provider/manual/manual.service.ts

```diff
@@ -220,39 +222,31 @@ export class ManualService implements DataProviderInterface {
     return { items };
   }
 
-  public async test(params: any) {
-    return this.scrape({
-      headers: params.headers,
-      selector: params.selector,
-      url: params.url
-    });
+  public async test(config: ScraperConfiguration) {
+    return this.scrape(config);
   }
 
-  private async scrape({
-    headers = {},
-    selector,
-    url
-  }: {
-    headers?: Headers;
-    selector: string;
-    url: string;
-  }): Promise<number> {
+  private async scrape(config: ScraperConfiguration): Promise<number> {
     try {
       const abortController = new AbortController();
 
       setTimeout(() => {
         abortController.abort();
       }, this.configurationService.get('REQUEST_TIMEOUT'));
 
-      const { body } = await got(url, {
-        headers,
+      const { body } = await got(config.url, {
+        headers: config.headers as Headers,
         // @ts-ignore
         signal: abortController.signal
       });
-
-      const $ = cheerio.load(body);
-
-      return extractNumberFromString($(selector).first().text());
+      if (config.type === 'json') {
```

Comment:
I would drop the type and check the content type of the response header instead. Something like:

```
const { body, headers } = await got(config.url, {
  headers: config.headers as Headers,
  // @ts-ignore
  signal: abortController.signal
});

if (headers["content-type"] === "application/json") {
  ...
} else {
  ...
}
```